drivers/s3.go

```diff
@@ -38,47 +38,51 @@ type s3OS struct {
 	awsAccessKeyID     string
 	awsSecretAccessKey string
 	s3svc              *s3.S3
+	gsSigner           *gsSigner
+	isGoogle           bool
 }
 
 type s3Session struct {
-	host           string
-	key            string
-	policy         string
-	signature      string
-	xAmzCredential string
-	xAmzDate       string
-	driver         *s3OS
+	host       string
+	key        string
+	policy     string
+	signature  string
+	credential string
+	xAmzDate   string
+	isGoogle   bool
 }
 
 // S3BUCKET s3 bucket owned by this node
 var S3BUCKET string
 
-// S3REGION region of s3 bucket owned by this node
-var S3REGION string
-
-func s3Host(bucket string) string {
+func s3Host(bucket string, isGoogle bool) string {
```

Comment:
We probably should just have a `googleHost` or `gsHost` function instead of overloading `s3Host`.